ui/packages/catalog-ui-search/src/main/webapp/react-component/location/base.line.js

```diff
@@ -12,204 +12,142 @@
-import styled from 'styled-components'
 
 const coordinatePairRegex = /-?\d{1,3}(\.\d*)?\s-?\d{1,3}(\.\d*)?/g
 
-const Invalid = styled.div`
-  background-color: ${props => props.theme.negativeColor};
-  height: 100%;
-  display: block;
-  overflow: hidden;
-  color: white;
-`
+function buildWktString(coordinates) {
+  return '[[' + coordinates.join('],[') + ']]'
+}
 
-class BaseLine extends React.Component {
-  invalidMessage = ''
-  constructor(props) {
-    super(props)
-    const { geometryKey } = props
-    const value = JSON.stringify(props[geometryKey])
-    this.state = { value, isValid: true }
-    this.state.isValid = true
-    this.is2DArray = this.is2DArray.bind(this)
-    this.validateListOfPoints = this.validateListOfPoints.bind(this)
-    this.isValidPolygon = this.isValidPolygon.bind(this)
-    this.isValidInput = this.isValidInput.bind(this)
-    this.removeErrorBox = this.removeErrorBox.bind(this)
+function convertWktString(value) {
+  if (value.includes('MULTI')) {
+    return convertMultiWkt(value.includes('POLYGON'), value)
+  } else if (value.includes('POLYGON') && value.endsWith('))')) {
+    return convertWkt(value, 4)
+  } else if (value.includes('LINESTRING') && value.endsWith(')')) {
+    return convertWkt(value, 2)
   }
-  componentWillReceiveProps(props) {
-    if (document.activeElement !== this.ref) {
-      const { geometryKey } = props
-      const value = JSON.stringify(props[geometryKey])
-      this.setState({ value })
-    }
+  return value
+}
+
+function convertWkt(value, numCoords) {
+  const coordinatePairs = value.match(coordinatePairRegex)
+  if (!coordinatePairs || coordinatePairs.length < numCoords) {
+    return value
+  }
+  const coordinates = coordinatePairs.map(coord => coord.replace(' ', ','))
+  return buildWktString(coordinates)
+}
+
+function convertMultiWkt(isPolygon, value) {
+  if (isPolygon && !value.endsWith(')))')) {
+    return value
+  } else if (!value.endsWith('))')) {
+    return value
   }
-  render() {
-    const props = this.props
-    const { label, cursor, geometryKey, unitKey, widthKey } = props
-    return (
-      <React.Fragment>
-        <div className="input-location">
+  const splitter = isPolygon ? '))' : ')'
+  const numPoints = isPolygon ? 4 : 2
+  let shapes = value
+    .split(splitter)
+    .map(shape => shape.match(coordinatePairRegex))
+  shapes = shapes
+    .filter(shape => shape !== null && shape.length >= numPoints)
+    .map(shape => shape.map(coordinatePair => coordinatePair.replace(' ', ',')))
+  return shapes.length === 0
+    ? value
+    : shapes.length === 1
+      ? buildWktString(shapes[0])
+      : '[' + shapes.map(shapeCoords => buildWktString(shapeCoords)) + ']'
+}
+
+const BaseLine = props => {
+  const {
+    label,
+    geometryKey,
+    setState,
+    setBufferState,
+    unitKey,
+    widthKey,
+    mode,
+    polyType,
+  } = props
+  const [currentValue, setCurrentValue] = useState(
+    JSON.stringify(props[geometryKey])
+  )
```

Comment:
turns out we don't actually need to maintain state for the current value, so this can be removed

Reply:
actually, after further testing, I take that back ☝️ 
If I'm editing a polygon that starts with `[[-97.438844,52.236694],` and I delete up to the decimal of either of those values, JSON.parse throws an error when attempting to set the state of the location-old model and the model isn't set. `currentValue` is, however, because it's set with the plain string. So the value state is necessary to maintain intermediary values that don't parse into JSONs when manually editing. Line and Polygon are unique in this sense because of their structure

Reply:
Eliminating the value state can be revisited now that the model is being set with invalid values